reproman/interface/jobs.py

```diff
@@ -72,11 +72,16 @@ def match(query_id, jobids):
 
 def _resurrect_orc(job):
     resource = get_manager().get_resource(job["resource_id"], "id")
-    with chpwd(job["local_directory"]):
-        orchestrator_class = ORCHESTRATORS[job["orchestrator"]]
-        orc = orchestrator_class(resource, job["submitter"], job,
-                                 resurrection=True)
-        orc.submitter.submission_id = job.get("_submission_id")
+    try:
+        with chpwd(job["local_directory"]):
+            orchestrator_class = ORCHESTRATORS[job["orchestrator"]]
+            orc = orchestrator_class(resource, job["submitter"], job,
+                                     resurrection=True)
+            orc.submitter.submission_id = job.get("_submission_id")
+    except FileNotFoundError:
```

Comment:
Just a note out loud- it is somewhat strange that `with` in python wasn't given `except` to handle Exceptions thrown with context manager itself (not the context block). Without it, there is no way to tell to handle only exceptions thrown within `chpwd` only. Or there is a general way for that?

Reply:
Thanks for raising this point.  One way to restrict the scope would be to instantiate the `chpwd` class separately.  I've pushed an update to do that.

<details>
<summary>range-diff</summary>

```
1:  f1220553d = 1:  f1220553d CLN: jobs: Drop unused exception binding
2:  e1f417ea8 = 2:  e1f417ea8 BF: jobs: Don't dump job spec when logging orchestrators errors
3:  b16279a61 = 3:  b16279a61 TST: run: Add local directory to context() fixture
4:  f2b372064 ! 4:  c7fdb3b03 BF: jobs: Treat deleted local directory as orchestrator error
    @@ reproman/interface/jobs.py: def match(query_id, jobids):
      def _resurrect_orc(job):
          resource = get_manager().get_resource(job["resource_id"], "id")
     -    with chpwd(job["local_directory"]):
    --        orchestrator_class = ORCHESTRATORS[job["orchestrator"]]
    --        orc = orchestrator_class(resource, job["submitter"], job,
    --                                 resurrection=True)
    --        orc.submitter.submission_id = job.get("_submission_id")
     +    try:
    -+        with chpwd(job["local_directory"]):
    -+            orchestrator_class = ORCHESTRATORS[job["orchestrator"]]
    -+            orc = orchestrator_class(resource, job["submitter"], job,
    -+                                     resurrection=True)
    -+            orc.submitter.submission_id = job.get("_submission_id")
    ++        # Create chpwd separately so that this try-except block doesn't cover
    ++        # the context manager suite below.
    ++        cd = chpwd(job["local_directory"])
     +    except FileNotFoundError:
     +        raise OrchestratorError(
     +            "local directory for job {} no longer exists: {}"
     +            .format(job["_jobid"], job["local_directory"]))
    -     return orc
    - 
    - 
    ++
    ++    with cd:
    +         orchestrator_class = ORCHESTRATORS[job["orchestrator"]]
    +         orc = orchestrator_class(resource, job["submitter"], job,
    +                                  resurrection=True)
     
      ## reproman/interface/tests/test_run.py ##
     @@
```

</details>